components/utils/ImageWithFallback.tsx

```diff
@@ -0,0 +1,40 @@
+'use client';
+
+import Image, { StaticImageData } from 'next/image';
+import { SyntheticEvent, useEffect, useState } from 'react';
+
+export interface ImageWithFallbackProps
+  extends React.ComponentProps<typeof Image> {
+  fallback: string | StaticImageData;
+  onError?: (e: SyntheticEvent<HTMLImageElement, Event>) => void;
+  [props: string]: any;
```

Comment:
why do we need `[props: string]: any`?

Reply:
but it's to general, can't we mention more explicit types?

Reply:
Oh, extends React.ComponentProps\<typeof Image> is the explicit types

Reply:
But still, what benefit does it give?

Reply:
Repeating my previous comments with a bit more verbosity might make the point clearer

```extends React.ComponentProps<typeof Image``` gives the benefit of having a type, such as intellisense
The props from ```[props: string]: any``` gives the benefit of actually using the props given to the component, by passing them to its child component ```</image>```

Reply:
and again buddy, if i remove only `[props: string]: any` i still get the intellisense benefit and nothing breaks so currently this part only confusing and doesn't give anything.
This is how it looks in my IDE -> 
```
interface ImageWithFallbackProps
  extends React.ComponentProps<typeof Image> {
  fallback: string | StaticImageData;
  onError?: (e: SyntheticEvent<HTMLImageElement, Event>) => void;
}
```
Also exporting this interface is redundant